examples/pwa/pwa.js

```diff
@@ -205,8 +205,11 @@ class AmpViewer {
     /** @private @const {...} */
     this.amp_ = null;
 
-    // Immediately install amp-shadow.js.
-    this.installScript_('/dist/amp-shadow.js');
+    // Install amp-shadow once our 3P polyfills are ready
+    this.win.addEventListener('WebComponentsReady', () => {
```

Comment:
So, no, not quite :) The goal is to start loading `amp-shadow.js` ASAP as well. If anything, I'd prefer if it was in the `<script async>` tags in the document's head. What we need to do in stead is convert "WebComponentsReady" to a promise that we use for document opening. Thus, we will have two promises:

1. `this.ampReadyPromise_` which we already have. It reports that AMP is fully ready to start opening documents.
2. We need shadowDomReadyPromise which will resolve when SD is ready to be used. It can be calculated similar to this:

```
shadowDomReadyPromise =
    Element.prototype.attachShadow || Element.prototype.createShadowRoot ?
    Promise.resolve() :   // Already there.
    new Promise(resolve => {   // Wait for polyfill to arrive
       this.win.addEventListener('WebComponentsReady', resolve);
    })
```

Then, document open will look simply like this:

```
Promise.all([this.ampReadyPromise_, shadowDomReadyPromise]).then(() => {
  AMP.attachShadowDocAsStream(...);
});
```